src/dotnet/ToolManifest/ToolManifestFile.cs

```diff
@@ -209,32 +258,22 @@ private SerializableLocalToolsManifest DeserializeLocalToolsManifest(FilePath po
             return result;
         }
 
-        private IEnumerable<(FilePath manifestfile, DirectoryPath manifestFileFirstEffectDirectory)> EnumerateDefaultAllPossibleManifests()
+        private IEnumerable<(FilePath manifestfile, DirectoryPath manifestFileFirstEffectDirectory)>
```

Comment:
Why do you even take this `manifestFileFirstEffectDirectory` parameter? It looks like it is not even being used.

Reply:
it is used to solve .config folder's "EffectDirectory". If a manifest file is nested under .config, .config should not be the first effect directory it should be it's parent.

.config/dotnet-tool.json
dotnet-tool.json

should be the "same" for the hierarchy